app/models/manager_refresh/inventory_collection.rb

```diff
@@ -124,6 +124,10 @@ def <<(inventory_object)
       end
     end
 
+    def build(hash)
+      self.<<(new_inventory_object(hash))
```

Comment:
I think, I'll add `.push` too and make this an alias. I found this a bit weird too